app/src/main/kotlin/androidx/compose/material3/NavigationDrawer2.kt

```diff
@@ -331,6 +314,90 @@
+    gesturesEnabled: Boolean = true,
+    scrimColor: Color = DrawerDefaults.scrimColor,
+    content: @Composable () -> Unit,
+) {
+    val scope = rememberCoroutineScope()
+    val density = LocalDensity.current
+    val minValue = with(density) { ContainerWidth.toPx() }
+    val maxValue = 0f
+
+    SideEffect {
+        drawerState.density = density
+        drawerState.anchoredDraggableState.updateAnchors(
+            DraggableAnchors {
+                DrawerValue.Closed at minValue
+                DrawerValue.Open at maxValue
+            },
+        )
+    }
+
+    val isRtl = LocalLayoutDirection.current == LayoutDirection.Rtl
+    val velocityTracker = remember { VelocityTracker() }
+    val dragModifier = if (gesturesEnabled) {
+        Modifier.pointerInput(isRtl) {
+            val multiplier = if (isRtl) -1 else 1
+            awaitEachGesture {
+                fun canConsume(slop: Float) = (slop > 0 && drawerState.isOpen) || (slop < 0 && drawerState.isClosed)
+                val down = awaitFirstDown(requireUnconsumed = false)
+                var overSlop: Float
+                val drag = awaitHorizontalTouchSlopOrCancellation(down.id) { change, over ->
+                    overSlop = over * multiplier
+                    if (canConsume(overSlop)) {
+                        velocityTracker.addPointerInputChange(change)
+                        change.consume()
+                        drawerState.anchoredDraggableState.dispatchRawDelta(overSlop)
+                    }
+                }
+                if (drag != null) {
+                    horizontalDrag(drag.id) {
+                        velocityTracker.addPointerInputChange(it)
+                        drag.consume()
+                        drawerState.anchoredDraggableState.dispatchRawDelta(it.positionChange().x * multiplier)
+                    }
+                    val velocity = velocityTracker.calculateVelocity()
+                    velocityTracker.resetTracking()
+                    scope.launch {
+                        drawerState.anchoredDraggableState.settle(velocity.x * multiplier)
+                    }
+                }
+            }
+        }
+    } else {
+        Modifier
+    }
+    Box(modifier.fillMaxSize().then(dragModifier)) {
+        Box {
+            content()
+        }
+        Scrim(
+            open = drawerState.isOpen,
+            onClose = {
+                if (gesturesEnabled && drawerState.confirmStateChange(DrawerValue.Closed)) {
+                    scope.launch { drawerState.close() }
+                }
+            },
+            fraction = {
+                calculateFraction(minValue, maxValue, drawerState.requireOffset())
+            },
+            color = scrimColor,
+        )
+        Box(
+            modifier = Modifier.offset {
+                IntOffset(drawerState.requireOffset().roundToInt(), 0)
+            }.align(Alignment.TopEnd).systemBarsPadding(),
```

Comment:
This doesn't look good in landscape mode, maybe we should pad the content instead.

Reply:
Done